pandas/core/base.py

```diff
@@ -554,7 +554,7 @@ def is_any_frame():
                                 name=getattr(self, 'name', None))
 
             return result, True
-        elif is_list_like(arg) and arg not in compat.string_types:
```

Comment:
Would still need this to be not `isinstance(arg, str)` I think

Reply:
Yeah, I meant to comment on this because it's a bit confusing.  What you're suggesting is what I think was the intention of the code, but isn't actually equivalent to what was previously written.

Substituting in the `PY3` version of `compat.string_types` yields:
```python
elif is_list_like(arg) and arg not in (str,):
```
This is slightly different than how we're interpreting the intention of the code:
```python
elif is_list_like(arg) and not isinstance(arg, str):
```
The first case `arg not in (str,)` is `False` for the `str` constructor and `True` for actual strings, and it's the opposite for `not isinstance(arg, str)`

Regardless of which is the correct, the second part of the conditional is redundant since `is_list_like` returns `False` in either case:

```python
In [2]: pd.core.dtypes.common.is_list_like(str)
Out[2]: False

In [3]: pd.core.dtypes.common.is_list_like('foo')
Out[3]: False
```

Reply:
Makes sense thanks for clarifying